### DISCUSSION TITLE
How to install a package with specific branch?

### DISCUSSION THREAD

**Discussion by adoyle-h** [mason-org/mason.nvim#672] [Q&A] [answered]
I want to install [bash-language-server](https://github.com/bash-lsp/bash-language-server/) with `main` branch.
But `:MasonInstall bash-language-server@main` failed.

```
  Failed
     bash-language-server
      Wrote to /Users/adoyle/.local/share/nvim/mason/.packages/bash-language-server/package.json:
      
      {
        "name": "@mason/bash-language-server",
        "version": "1.0.0",
        "description": "",
        "main": "index.js",
        "scripts": {
          "test": "echo \"Error: no test specified\" && exit 1"
        },
        "keywords": [],
        "author": "",
        "license": "ISC"
      }
      
      
      npm ERR! code ETARGET
      npm ERR! notarget No matching version found for bash-language-server@main.
      npm ERR! notarget In most cases you or one of your dependencies are requesting
      npm ERR! notarget a package version that doesn't exist.
      
      npm ERR! A complete log of this run can be found in:
      npm ERR!     /Users/adoyle/.npm/_logs/2022-11-17T09_30_33_315Z-debug-0.log
      spawn: npm failed with exit code 1 and signal 0. 
```

**Comment by yutkat** [comment, marked as answer]
Because it is installed from npm(https://www.npmjs.com/package/bash-language-server). You can switch npm versions with @xxx(https://www.npmjs.com/package/bash-language-server?activeTab=versions).

If you want to build from source on the main branch, you'll have to build it yourself.
https://github.com/bash-lsp/bash-language-server/blob/master/docs/development-guide.md#initial-setup

**Reply by adoyle-h** [reply]
If I build a npm package from source, how to install it via Mason?

**Reply by yutkat** [reply]
I think it would be better to release a new version via npm, but this way you can create the package manually.
https://github.com/williamboman/mason.nvim/discussions/603#discussioncomment-3966410
